src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -232,9 +247,11 @@ public ICompilationTracker FreezePartialStateWithTree(SolutionState solution, Do
                 out CompilationPair compilations,
                 out CompilationTrackerGeneratorInfo generatorInfo,
                 out Dictionary<MetadataReference, ProjectId>? metadataReferenceToProjectId,
+                out CompilationTrackerState oldStateForBugInvestigation,
                 CancellationToken cancellationToken)
             {
                 var state = ReadState();
+                oldStateForBugInvestigation = state;
```

Comment:
Capturing this value here now, because otherwise the failures I see are after this function has ran, and another thread could have updated this state.